sunpy/map/mapbase.py

```diff
@@ -221,6 +221,12 @@ def __repr__(self):
            dim=u.Quantity(self.dimensions),
            scale=u.Quantity(self.scale),
            tmf=TIME_FORMAT) + self.data.__repr__())
+                                                                        
+    def _modified_copy(self, data, meta, plot_settings=None, **kwargs):
+        """
+        Method to create new modified map object with the required changes
+        """
+        return GenericMap(data, meta, **kwargs)
```

Comment:
This is the root of the test failure. You need to return the same type as `self` not just a `GenericMap` to fix the test you need to replace this by:

```python
new_map = type(self)(data, meta, **kwargs)
return new_map
```

Also you need to set plot settings in this function:

```python
new_map = type(self)(data, meta, **kwargs)
if plot_settings:
    new_map.plot_settings = plot_settings
return new_map
```